prediction_market_agent/agents/microchain_agent/app_nft_game.py

```diff
@@ -0,0 +1,156 @@
+"""
+PYTHONPATH=. streamlit run prediction_market_agent/agents/microchain_agent/app_nft_game.py
+
+Tip: if you specify PYTHONPATH=., streamlit will watch for the changes in all files, instead of just this one.
+"""
+
+import typing as t
+
+import streamlit as st
+from prediction_market_agent_tooling.tools.balances import get_balances
+from prediction_market_agent_tooling.tools.datetime_utc import DatetimeUTC
+
+from prediction_market_agent.agents.identifiers import AgentIdentifier
+from prediction_market_agent.agents.microchain_agent.agents_nft_game import (
+    AGENTS_IN_THE_GAME,
+    TREASURY_SAFE_ADDRESS,
+    NFTAgent,
+)
+from prediction_market_agent.agents.microchain_agent.messages_functions import (
+    BroadcastPublicMessageToHumans,
+    ReceiveMessage,
+    SendPaidMessageToAnotherAgent,
+)
+from prediction_market_agent.db.long_term_memory_table_handler import (
+    LongTermMemoryTableHandler,
+)
+from prediction_market_agent.db.prompt_table_handler import PromptTableHandler
+
+st.set_page_config(
+    page_title="Agent's NFT-locked Treasury Game", page_icon="🎮", layout="wide"
+)
+
+
+@st.cache_resource
+def long_term_memory_table_handler(
+    identifier: AgentIdentifier,
+) -> LongTermMemoryTableHandler:
+    return LongTermMemoryTableHandler.from_agent_identifier(identifier)
+
+
+@st.cache_resource
+def prompt_table_handler(identifier: AgentIdentifier) -> PromptTableHandler:
+    return PromptTableHandler.from_agent_identifier(identifier)
+
+
+def send_message_part(nft_agent: NFTAgent) -> None:
+    message = st.text_area("Write a message to the agent")
+
+    if st.button("Send message", disabled=not message):
+        # TODO: Don't do this manually with deployment private key, use the user's wallet!
```

Comment:
Created issue for this here https://github.com/gnosis/prediction-market-agent/issues/591

Turns out https://github.com/AlgoveraAI/streamlit-metamask is meant only for ERC-20 transfers (and didn't work on first try anyway 😄 ).

But I don't want to block the showcasing of this because of that, we can use this to show it off.

Reply:
I was hoping the streamlit-metamask extension was gonna work like a charm :(

Anyways, what do you think about adding 1 text input (RPC_URL) so that we are able to send transactions on a local chain, instead of burning xDAI on Mainnet during the demo?

<img width="1585" alt="image" src="https://github.com/user-attachments/assets/68ff07d3-f118-4e00-85a1-8b8ac1e3ace5">

Reply:
We are able to change the `GNOSIS_RPC_URL` variable in the deployment's environment. It doesn't help to do it  only here if all the agents aren't deployed on the same local chain.

Reply:
`GNOSIS_RPC_URL` is probably good enough for now - I want to avoid doing demos, etc on the main chain, but I have other ideas. This can be closed.